spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala

```diff
@@ -400,52 +466,98 @@ class CometSparkSessionExtensions
                 op.right,
                 SerializedPlan(None))
             case None =>
-              op
+              opWithInfo(op, info)
           }
 
         case c @ CoalesceExec(numPartitions, child)
             if isCometOperatorEnabled(conf, "coalesce")
               && isCometNative(child) =>
-          QueryPlanSerde.operator2Proto(c) match {
+          val (newOp, info) = QueryPlanSerde.operator2Proto(c)
+          newOp match {
             case Some(nativeOp) =>
               val cometOp = CometCoalesceExec(c, numPartitions, child)
               CometSinkPlaceHolder(nativeOp, c, cometOp)
             case None =>
-              c
+              opWithInfo(c, info)
           }
 
         case s: TakeOrderedAndProjectExec
             if isCometNative(s.child) && isCometOperatorEnabled(conf, "takeOrderedAndProjectExec")
               && isCometShuffleEnabled(conf) &&
-              CometTakeOrderedAndProjectExec.isSupported(s) =>
-          QueryPlanSerde.operator2Proto(s) match {
+              CometTakeOrderedAndProjectExec
+                .isSupported(s)
+                ._1 =>
+          // TODO: support offset for Spark 3.4
```

Comment:
Do we need to file an issue to track this?

Reply:
This got copied from some other branch so I've removed it. I'm not too sure what else we need to support for `CometTakeOrderedAndProjectExec` but let me find out and log an issue